### PR TITLE
Add startActivityForResult explanation

### PR DESCRIPTION
**Pull request by SUPERCILEX**
#434

### CODE REVIEW

File: auth/README.md [hunk 1]
```diff
@@ -91,14 +91,15 @@ Twitter app as reported by the [Twitter application manager](https://apps.twitte
 </resources>
 ```
 
-In addition, if are using Smart Lock or require a user's email, you must enable the
+In addition, if you are using Smart Lock or require a user's email, you must enable the
```
Comment (SUPERCILEX) [inline auth/README.md]: Whoops

File: auth/README.md [hunk 2]
```diff
@@ -147,6 +148,14 @@ startActivityForResult(
     RC_SIGN_IN);
 ```
 
+<a href="https://developer.android.com/reference/android/app/Activity.html#startActivityForResult(android.content.Intent, int)">`startActivityForResult`</a>
```
Comment (SUPERCILEX) [inline auth/README.md]: This needs to be in HTML because of the parentheses in the link.

File: auth/README.md [hunk 3]
```diff
@@ -147,6 +148,14 @@ startActivityForResult(
...
+is part of the Android framework and takes two parameters: an Intent to start an activity--in this case,
+the sign in builder--and a request code of type `int`. For example, `REQUEST_CODE_SIGN_IN` or `RC_SIGN_IN`
+for short is a constant defined in our activity: `private static final int RC_SIGN_IN = 100;`.
+The number you chose is arbitrary as long as it doesn't conflict with other
+`startActivityForResult` requests you might be making. In [response codes](#response-codes),
+you will see how we can use `RC_SIGN_IN` to differentiate between multiple `startActivityForResult` requests.
+
```
Comment (SUPERCILEX) [inline auth/README.md]: This is really meaty, what do you think?

Comment (samtstern) [inline auth/README.md]: I think that discusses the mechanics of startActivityForResult a little too much, how about something like this:

To kick off the FirebaseUI sign in flow, call `startActivityForResult(...)` on the sign in Intent you built.  The second parameter (`RC_SIGN_IN`) is a request code you define to identify the request when the result is returned to your app in `onActivityResult(...)`.  See the [response codes](#response-codes) section below for more details on receiving the results of the sign in flow.

File: auth/README.md [hunk 4]
```diff
@@ -216,28 +225,30 @@ supported.
...
-        startActivity(new Intent(this, WelcomeBackActivity.class));
-        finish();
-        return;
-    }
-
-    // Sign in canceled
-    if (resultCode == RESULT_CANCELED) {
-        showSnackbar(R.string.sign_in_cancelled);
-        return;
+    if (requestCode == RC_SIGN_IN) {
```
Comment (samtstern) [inline auth/README.md]: Add a small comment here something like `// RC_SIGN_IN is the request code you passed into startActivityForResult when starting the sign in flow`

File: auth/README.md [hunk 5]
```diff
@@ -147,6 +148,14 @@ startActivityForResult(
     RC_SIGN_IN);
```
Comment (samtstern) [inline auth/README.md]: We should add a line above this code snippet to be super clear:

```java

// Choose an arbitrary request code value
private static final int RC_SIGN_IN = 123;

// ...

  startActivityForResult(
```

**Comment by samtstern** [comment]
Thanks for getting on this so fast!

File: auth/README.md (hunk 3 above)
Comment (SUPERCILEX) [inline auth/README.md]: Yup, I went a bit too far... 😄

**Comment by SUPERCILEX** [comment]
Done!

**Pull request merged**